test/__snapshots__/test.js.snap

```diff
@@ -2983,12 +3023,55 @@ exports.install = (globalObject, globalNames) => {
       }
       return esValue[implSymbol].promiseConsumer(...args);
     }
+
+    async promiseOperation() {
+      const esValue = this !== null && this !== undefined ? this : globalObject;
+      if (!exports.is(esValue)) {
+        throw new TypeError(\\"Illegal invocation\\");
+      }
+
+      return utils.tryWrapperForImpl(esValue[implSymbol].promiseOperation());
```

Comment:
This has to generate:
```js
       return utils.tryWrapperForImpl(await esValue[implSymbol].promiseOperation());
```

Reply:
This and your other unsolicited comments are not so straightforward.

You are suggesting that we incorporate the unwrapping done in https://github.com/jsdom/webidl2js/pull/108 here. However, that breaks the ability for the impl method/getter to return the same promise multiple times. That is especially problematic for attributes, where `x.attr === x.attr` is almost always what the spec intends.

I think we should leave working on the unwrapping to https://github.com/jsdom/webidl2js/pull/108 or some continuation of that has better discussion behind it.